compiler-core/src/parse/error.rs

```diff
@@ -397,3 +405,14 @@ impl LexicalError {
         }
     }
 }
+
+impl fmt::Display for UnexpectedTokenInfo {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let base = self.display.replace("\"", "");
+        if self.is_keyword {
+            write!(f, "the keyword `{}`", base)
+        } else {
+            write!(f, "\"{}\"", base)
+        }
+    }
+}
```

Comment:
Do not implement this using traits please. We avoid using traits, and this specifically is a misuse of Display as this isn't a general purpose to-string impl, it's specific to that one function.